src/client/Microsoft.Identity.Client/Region/RegionDiscoveryProvider.cs

```diff
@@ -21,18 +24,15 @@ internal sealed class RegionDiscoveryProvider : IRegionDiscoveryProvider
         private const string RegionName = "REGION_NAME";
 
         // For information of the current api-version refer: https://docs.microsoft.com/en-us/azure/virtual-machines/windows/instance-metadata-service#versioning
-        private readonly Uri _ImdsUri = new Uri("http://169.254.169.254/metadata/instance/compute?api-version=2020-06-01");
+        private const string ImdsEndpoint = "http://169.254.169.254/metadata/instance/compute";
+        private const string DefaultApiVersion = "2020-06-01";
 
-        private IDictionary<string, string> Headers;
         private readonly IHttpManager _httpManager;
-        private static INetworkCacheMetadataProvider _networkCacheMetadataProvider;
+        private readonly INetworkCacheMetadataProvider _networkCacheMetadataProvider;
 
         public RegionDiscoveryProvider(IHttpManager httpManager, INetworkCacheMetadataProvider networkCacheMetadataProvider = null)
```

Comment:
The logic here is important, it orchestrates the entire regional discovery flow which you have worked on. I would add some logging so that you can understand what's happening when things fail.

Reply:
Added more logging